test-util/src/main/java/com/linkedin/test/util/VariableRetries.java

```diff
@@ -0,0 +1,47 @@
+/*
+   Copyright (c) 2020 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.test.util;
+
+import org.testng.IRetryAnalyzer;
+import org.testng.ITestResult;
+
+
+/**
+ * Allows N retries for a given test method. Subclass implementations must specify the value of N.
+ *
+ * Note that the same instance is used for all iterations of a test method, meaning that even if there are multiple
+ * iterations (e.g. data provider provides multiple sets of input) only N retries will be allowed.
+ *
+ * @author Evan Williams
+ */
+public abstract class VariableRetries implements IRetryAnalyzer
```

Comment:
I am afraid I am not a fan of this naming. How about just `Retires` or `RetrySetup` ?

Reply:
Haha I agree, I originally named it `NRetries` but I thought it looked weird. I can just do `Retries` or something like that.